### ISSUE TITLE
`const_constructor_param_type_mismatch`: False positive when using ternary expression with the `identical` function in constant constructor

### ISSUE THREAD

**Issue by hacker1024** [dart-lang/sdk#48127]
**Describe the issue**
When using a ternary expression with the `identical` function inside a constant constructor, the `const_constructor_param_type_mismatch` lint is triggered.

This also occurs when using a constant boolean with a value calculated using `identical`, like Flutter's `kIsWeb`.

**To Reproduce**
Analyse the following (or just paste it into [DartPad](http://dartpad.dev)). Note that a `const_constructor_param_type_mismatch` error is reported, but the code compiles and runs just fine.

```dart
class ExampleClass {
  const ExampleClass.string(String value);
  const ExampleClass.integer(int value);
  const ExampleClass.boolean(bool value);
}

// error: A value of type 'Null' can't be assigned to a parameter of type 'String' in a const constructor.
const test1 = ExampleClass.string(identical(0, 0.0) ? 'Yes' : 'No');

// error: A value of type 'Null' can't be assigned to a parameter of type 'int' in a const constructor.
const test2 = ExampleClass.integer(identical(0, 0.0) ? 1 : 0);

// This, however, is fine, as there's no ternary expression:
const test3 = ExampleClass.boolean(identical(0, 0.0));

void main() => print('$test1, $test2, $test3');
```

**Expected behavior**
No errors should be reported during analysis.

**Additional context**

**Comment by srawlins** [comment]
I no longer see this error, and I know @scheglov fixed one or two `kIsWeb` issues recently. Please re-open if you can reproduce this at DartPad today (Dart 2.16) or otherwise with recent Dart.

**Comment by hacker1024** [comment]
This issue does seem to be fixed in Dart 2.16. Thanks!

**Comment by hacker1024** [comment]
For those curious, this seems to be the relevant fix: https://github.com/dart-lang/sdk/commit/d09da53c116241a2f710dd11a31deebb14a244c0
